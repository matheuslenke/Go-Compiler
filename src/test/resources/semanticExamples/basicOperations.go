package samples

func testeOperacao() {
	var int1 int = -1 // +
	var int2 int = 300

	var add int = int1 + int2
	var sub int = int1 - int2
	var mul int = int1 * int2
	var div int = int1 / int2

	var eq bool = int1 == int2
	var lt bool = int1 < int2
	var gt bool = int1 > int2
	var let bool = int1 <= int2
	var get bool = int1 >= int2

	var float1 float32 = 15.0
	var float2 float32 = 20.0

	var add2 float32 = float1 + float2
	var sub2 float32 = float1 - float2
	var mul2 float32 = float1 * float2
	var div2 float32 = float1 / float2

	var string1 string = "Hello"
	var string2 string = "World"

	var add3 string = string1 + string2

}
