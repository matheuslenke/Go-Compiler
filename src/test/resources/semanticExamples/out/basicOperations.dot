digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="(func) testeOperacao@0"];
node2[label="parameters"];
node1 -> node2;
node3[label="block"];
node4[label="var_list"];
node5[label="="];
node6[label="(int) int1@1"];
node5 -> node6;
node7[label="(int) int@1"];
node5 -> node7;
node4 -> node5;
node3 -> node4;
node8[label="var_list"];
node9[label="="];
node10[label="(int) int2@2"];
node9 -> node10;
node11[label="(int) int@300"];
node9 -> node11;
node8 -> node9;
node3 -> node8;
node12[label="var_list"];
node13[label="="];
node14[label="(int) add@3"];
node13 -> node14;
node15[label="(int) +"];
node16[label="(int) int1@1"];
node15 -> node16;
node17[label="(int) int2@2"];
node15 -> node17;
node13 -> node15;
node12 -> node13;
node3 -> node12;
node18[label="var_list"];
node19[label="="];
node20[label="(int) sub@4"];
node19 -> node20;
node21[label="(int) -"];
node22[label="(int) int1@1"];
node21 -> node22;
node23[label="(int) int2@2"];
node21 -> node23;
node19 -> node21;
node18 -> node19;
node3 -> node18;
node24[label="var_list"];
node25[label="="];
node26[label="(int) mul@5"];
node25 -> node26;
node27[label="(int) *"];
node28[label="(int) int1@1"];
node27 -> node28;
node29[label="(int) int2@2"];
node27 -> node29;
node25 -> node27;
node24 -> node25;
node3 -> node24;
node30[label="var_list"];
node31[label="="];
node32[label="(int) div@6"];
node31 -> node32;
node33[label="(int) /"];
node34[label="(int) int1@1"];
node33 -> node34;
node35[label="(int) int2@2"];
node33 -> node35;
node31 -> node33;
node30 -> node31;
node3 -> node30;
node36[label="var_list"];
node37[label="="];
node38[label="(bool) eq@7"];
node37 -> node38;
node39[label="(bool) =="];
node40[label="(int) int1@1"];
node39 -> node40;
node41[label="(int) int2@2"];
node39 -> node41;
node37 -> node39;
node36 -> node37;
node3 -> node36;
node42[label="var_list"];
node43[label="="];
node44[label="(bool) lt@8"];
node43 -> node44;
node45[label="(bool) <"];
node46[label="(int) int1@1"];
node45 -> node46;
node47[label="(int) int2@2"];
node45 -> node47;
node43 -> node45;
node42 -> node43;
node3 -> node42;
node48[label="var_list"];
node49[label="="];
node50[label="(bool) gt@9"];
node49 -> node50;
node51[label="(bool) >"];
node52[label="(int) int1@1"];
node51 -> node52;
node53[label="(int) int2@2"];
node51 -> node53;
node49 -> node51;
node48 -> node49;
node3 -> node48;
node54[label="var_list"];
node55[label="="];
node56[label="(bool) let@10"];
node55 -> node56;
node57[label="(bool) <="];
node58[label="(int) int1@1"];
node57 -> node58;
node59[label="(int) int2@2"];
node57 -> node59;
node55 -> node57;
node54 -> node55;
node3 -> node54;
node60[label="var_list"];
node61[label="="];
node62[label="(bool) get@11"];
node61 -> node62;
node63[label="(bool) >="];
node64[label="(int) int1@1"];
node63 -> node64;
node65[label="(int) int2@2"];
node63 -> node65;
node61 -> node63;
node60 -> node61;
node3 -> node60;
node66[label="var_list"];
node67[label="="];
node68[label="(float) float1@12"];
node67 -> node68;
node69[label="(float) real@15.0"];
node67 -> node69;
node66 -> node67;
node3 -> node66;
node70[label="var_list"];
node71[label="="];
node72[label="(float) float2@13"];
node71 -> node72;
node73[label="(float) real@20.0"];
node71 -> node73;
node70 -> node71;
node3 -> node70;
node74[label="var_list"];
node75[label="="];
node76[label="(float) add2@14"];
node75 -> node76;
node77[label="(float) +"];
node78[label="(float) float1@12"];
node77 -> node78;
node79[label="(float) float2@13"];
node77 -> node79;
node75 -> node77;
node74 -> node75;
node3 -> node74;
node80[label="var_list"];
node81[label="="];
node82[label="(float) sub2@15"];
node81 -> node82;
node83[label="(float) -"];
node84[label="(float) float1@12"];
node83 -> node84;
node85[label="(float) float2@13"];
node83 -> node85;
node81 -> node83;
node80 -> node81;
node3 -> node80;
node86[label="var_list"];
node87[label="="];
node88[label="(float) mul2@16"];
node87 -> node88;
node89[label="(float) *"];
node90[label="(float) float1@12"];
node89 -> node90;
node91[label="(float) float2@13"];
node89 -> node91;
node87 -> node89;
node86 -> node87;
node3 -> node86;
node92[label="var_list"];
node93[label="="];
node94[label="(float) div2@17"];
node93 -> node94;
node95[label="(float) /"];
node96[label="(float) float1@12"];
node95 -> node96;
node97[label="(float) float2@13"];
node95 -> node97;
node93 -> node95;
node92 -> node93;
node3 -> node92;
node98[label="var_list"];
node99[label="="];
node100[label="(string) string1@18"];
node99 -> node100;
node101[label="(string) string@0"];
node99 -> node101;
node98 -> node99;
node3 -> node98;
node102[label="var_list"];
node103[label="="];
node104[label="(string) string2@19"];
node103 -> node104;
node105[label="(string) string@1"];
node103 -> node105;
node102 -> node103;
node3 -> node102;
node106[label="var_list"];
node107[label="="];
node108[label="(string) add3@20"];
node107 -> node108;
node109[label="(string) +"];
node110[label="(string) string1@18"];
node109 -> node110;
node111[label="(string) string2@19"];
node109 -> node111;
node107 -> node109;
node106 -> node107;
node3 -> node106;
node1 -> node3;
node0 -> node1;
}
