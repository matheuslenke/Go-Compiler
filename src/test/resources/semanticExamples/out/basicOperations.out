[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:




Variables table:
Entry 0 -- name: testeOperacao, line: 3, type: func, scopeName: global
Entry 1 -- name: int1, line: 4, type: int, scopeName: testeOperacao
Entry 2 -- name: int2, line: 5, type: int, scopeName: testeOperacao
Entry 3 -- name: add, line: 7, type: int, scopeName: testeOperacao
Entry 4 -- name: sub, line: 8, type: int, scopeName: testeOperacao
Entry 5 -- name: mul, line: 9, type: int, scopeName: testeOperacao
Entry 6 -- name: div, line: 10, type: int, scopeName: testeOperacao
Entry 7 -- name: eq, line: 12, type: bool, scopeName: testeOperacao
Entry 8 -- name: lt, line: 13, type: bool, scopeName: testeOperacao
Entry 9 -- name: gt, line: 14, type: bool, scopeName: testeOperacao




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.793 s
[INFO] Finished at: 2022-02-24T20:53:44-03:00
[INFO] ------------------------------------------------------------------------
